src/components/repository/RepositoryViewBase/RepositoryViewBase.vue

```diff
@@ -38,29 +68,49 @@
         <bh-loading />
       </div>
     </div>
+    <train-modal
+      v-if="repository"
+      :training="training"
+      :ready-for-train="repository.ready_for_train"
+      :requirements-to-train="repository.requirements_to_train"
+      :languages-ready-for-train="repository.languages_ready_for_train"
+      :open.sync="trainModalOpen"
+      @train="train()" />
+    <train-response
+      v-if="trainResponseData"
+      :train-response="trainResponseData"
+      :open="openTrainResponse"
+      @closeTrainResponse="openTrainResponse = false" />
```

Comment:
You don't need have a event to close this modal.. Send from `train-response` synced event to update open prop.